library/L1_Peripheral/lpc17xx/spi_bus.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "L1_Peripheral/lpc17xx/pin.hpp"
+#include "L1_Peripheral/lpc17xx/spi.hpp"
```

Comment:
Which one would I include in order to use SPI? As it looks here, I have to `#include "spi_bus.hpp"` if that is the case, it would be better to swap names, having this become `spi.hpp` and have the other be `spi_bus.hpp`

Reply:
To use SPI, `spi_bus.hpp` should be included. For some reason swapping the names would cause the errors to show up.

Reply:
Ok, for now, lets pull in all of the changes that do not require the `_bus.hpp` hack to work. We can then spend some time to figure out why the naming is so fragile. Because allowing this problem to stay this will cause more headache down the road.